### PR TITLE
Improvements and more

### PR DESCRIPTION
**Pull request by luizfernandonb**
`1.` - `README.md` has been modified to be more user-friendly
`2.` - `GDB` has been updated to version 10.1 (the most recent so far)
`3.` - Instead of telling the user that it was not possible to create the `$PS3DEV` folder, the toolchain will create it and give the user ownership.
`4.` - Some files that were downloaded in `.bz2` and `.xz` were modified to be downloaded in `.tar.gz` and with that it was possible to use the `pigz` program that uses multi-thread to decompress `.tar.gz` files
`5.` - New flags in the compilation were added (I was a little hesitant to use the `-O3` flag but decided to use it anyway), if they don't think it's good, I can remove

In short this is it, if you look at what has been modified, you will see it was practically this

### CODE REVIEW

File: depends/check-ps3dev.sh [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
+#
 # check-ps3dev.sh by Naomi Peori (naomi@peori.ca)
+# Modified by luizfernandonb (luizfernando.nb@outlook.com)
 
 ## Check if $PS3DEV is set.
 if test ! $PS3DEV; then { echo "ERROR: Set \$PS3DEV before continuing."; exit 1; } fi
 
 ## Check for the $PS3DEV directory.
-( ls -ld $PS3DEV || mkdir -p $PS3DEV ) 1>/dev/null 2>&1 || { echo "ERROR: Create $PS3DEV before continuing."; exit 1; }
+( ls -ld $PS3DEV || mkdir -p $PS3DEV ) 1>/dev/null 2>&1 || { sudo mkdir $PS3DEV && sudo chown $USER:$USER $PS3DEV; } # { echo "ERROR: Create $PS3DEV before continuing."; exit 1; }
```
Comment (zeldin) [inline depends/check-ps3dev.sh:10]: Eh, no.  _Never_ use sudo in scripts.

File: depends/check-ps3dev.sh [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
 
 ## Check for write permission.
-touch $PS3DEV/test.tmp 1>/dev/null 2>&1 || { echo "ERROR: Grant write permissions for $PS3DEV before continuing."; exit 1; }
+# touch $PS3DEV/test.tmp 1>/dev/null 2>&1 || { echo "ERROR: Grant write permissions for $PS3DEV before continuing."; exit 1; }
```
Comment (zeldin) [inline depends/check-ps3dev.sh:13]: Why was this check removed?

File: scripts/001-binutils-PPU.sh [hunk 3]
```diff
@@ -1,19 +1,21 @@
...
 # binutils-PPU.sh by Naomi Peori (naomi@peori.ca)
+# Modified by luizfernandonb (luizfernando.nb@outlook.com)
 
 BINUTILS="binutils-2.22"
 
 if [ ! -d ${BINUTILS} ]; then
 
   ## Download the source code.
-  if [ ! -f ${BINUTILS}.tar.bz2 ]; then wget --continue https://ftp.gnu.org/gnu/binutils/${BINUTILS}.tar.bz2; fi
+  if [ ! -f ${BINUTILS}.tar.gz ]; then wget --continue https://ftp.gnu.org/gnu/binutils/${BINUTILS}.tar.gz; fi
```
Comment (zeldin) [inline scripts/001-binutils-PPU.sh:11]: Why?  This just makes the download 34% larger.

File: scripts/001-binutils-PPU.sh [hunk 4]
```diff
@@ -1,19 +1,21 @@
...
 
   ## Download an up-to-date config.guess and config.sub
   if [ ! -f config.guess ]; then wget --continue https://git.savannah.gnu.org/cgit/config.git/plain/config.guess; fi
   if [ ! -f config.sub ]; then wget --continue https://git.savannah.gnu.org/cgit/config.git/plain/config.sub; fi
 
   ## Unpack the source code.
-  tar xfvj ${BINUTILS}.tar.bz2
+  pigz -dc ${BINUTILS}.tar.gz | tar -xvf -
```
Comment (zeldin) [inline scripts/001-binutils-PPU.sh:18]: And what in the world is `pigz`?  Only use tools people are likely to actually have...

File: scripts/005-binutils-SPU.sh [hunk 5]
```diff
@@ -42,7 +44,13 @@ cd ${BINUTILS}/build-spu
...
     --with-gcc \
     --with-gnu-as \
-    --with-gnu-ld
+    --with-gnu-ld \
+    CFLAGS="-O3" \
+    CXXFLAGS="-g -O3" \
+    CCFLAGS_FOR_TARGET="-g -O3" \
+    CXXFLAGS_FOR_TARGET="-g -O3" \
+    GOCFLAGS_FOR_TARGET="-O3 -g" \
+    BOOT_CFLAGS="-g -O3" \
```
Comment (shagkur) [inline scripts/005-binutils-SPU.sh:54]: Dont use -O3 here. SPU code needs to be as samll as possible. -Os and a few more other options are to go here for the TARGET flags. '-g' is already passed by default. And as a personal note: I prefer to have the flags before the '../configure' call.

File: scripts/006-gcc-newlib-SPU.sh [hunk 6]
```diff
@@ -1,22 +1,25 @@
 #!/bin/sh -e
+#
 # gcc-newlib-SPU.sh by Naomi Peori (naomi@peori.ca)
+# Modified by luizfernandonb (luizfernando.nb@outlook.com)
 
 GCC="gcc-7.2.0"
 NEWLIB="newlib-1.20.0"
+PS3="-PS3"
```
Comment (shagkur) [inline scripts/006-gcc-newlib-SPU.sh:9]: Don't get it why we need an extra variable here? These scripts only build for PS3, so no need imo, to add another "indirection" here.

**Review by shagkur — CHANGES_REQUESTED** [review]

**Comment by zeldin** [comment]
> `4.` - Some files that were downloaded in `.bz2` and `.xz` were modified to be downloaded in `.tar.gz` and with that it was possible to use the `pigz` program that uses multi-thread to decompress `.tar.gz` files

Dude.  You increased the download time of binutils by 12 seconds in order to speed up an uncompress which takes 4 seconds...  😆

File: scripts/001-binutils-PPU.sh (hunk 3 above)
Comment (shagkur) [reply scripts/001-binutils-PPU.sh:11]: Didn't get this change either. Just for the speed improvement of decompression? No need to change this here. :+1:

File: scripts/001-binutils-PPU.sh (hunk 4 above)
Comment (shagkur) [reply scripts/001-binutils-PPU.sh:18]: :+1:

File: depends/check-ps3dev.sh (hunk 1 above)
Comment (luizfernandonb) [reply depends/check-ps3dev.sh:10]: Why not?

File: depends/check-ps3dev.sh (hunk 2 above)
Comment (luizfernandonb) [reply depends/check-ps3dev.sh:13]: If the owner of the folder is already the [user himself](https://github.com/ps3dev/ps3toolchain/pull/106/files/c435cf5a628d215c26e6137fbc5515ae41aecb03#diff-c2959f84736240516e7fa192da534108feb8b78da44178e9fadd842f9e133446R10), there is no need to check if he can write to the folder

File: scripts/001-binutils-PPU.sh (hunk 3 above)
Comment (luizfernandonb) [reply scripts/001-binutils-PPU.sh:11]: The compilation of the toolchain is already very slow as a whole, the `pigz` (tool that only decompresses .tar.gz) speeds up the decompression a little more, it's not much, but it helps, but if you want I can remove the difference in time it wasn't that big between the `pigz` and the `tar` it wasn't that big

File: scripts/001-binutils-PPU.sh (hunk 4 above)
Comment (luizfernandonb) [reply scripts/001-binutils-PPU.sh:18]: If you read the [README](https://github.com/ps3dev/ps3toolchain/pull/106/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R31) changes you know that `pigz` is in the libraries that are asked to install before compiling, I would not use a program in the script file without even installing it

File: scripts/005-binutils-SPU.sh (hunk 5 above)
Comment (luizfernandonb) [reply scripts/005-binutils-SPU.sh:54]: Why does the SPU code need to be as small as possible? So I believe it is best to leave it in the default

File: scripts/006-gcc-newlib-SPU.sh (hunk 6 above)
Comment (luizfernandonb) [reply scripts/006-gcc-newlib-SPU.sh:9]: Just to make it easier to do maintenance if you ever need to change the script

**Comment by luizfernandonb** [comment]
> > `4.` - Some files that were downloaded in `.bz2` and `.xz` were modified to be downloaded in `.tar.gz` and with that it was possible to use the `pigz` program that uses multi-thread to decompress `.tar.gz` files
> 
> Dude. You increased the download time of binutils by 12 seconds in order to speed up an uncompress which takes 4 seconds... 😆

For me the download time stayed the same 🤔

File: scripts/005-binutils-SPU.sh (hunk 5 above)
Comment (shagkur) [reply scripts/005-binutils-SPU.sh:54]: Because on each SPU you only have 256Kb local store memory, which is used for executable code, program data and stack. So if the executable code gets big you don't have much left to operate on with.

File: scripts/006-gcc-newlib-SPU.sh (hunk 6 above)
Comment (shagkur) [reply scripts/006-gcc-newlib-SPU.sh:9]: i still dont see the advantage. All you'll have to change here is probably the version of newlib and gcc. The "PS3" is only used for the patch scripts which is fully under our control.

File: scripts/005-binutils-SPU.sh (hunk 5 above)
Comment (luizfernandonb) [reply scripts/005-binutils-SPU.sh:54]: Ah yes, sorry, I forgot that detail

**Comment by luizfernandonb** [comment]
I believe that everything is now all right

**Comment by miigotu** [comment]
Is it just me or does this feel like a lot of unnecessary changes for a set of scripts that were slowly refined over like 10 years?

File: depends/check-ps3dev.sh (hunk 1 above)
Comment (zeldin) [reply depends/check-ps3dev.sh]: Privilege separation exists for a reason.  Escalation of privilege must always be a conscious decision by the operator.

File: depends/check-ps3dev.sh (hunk 2 above)
Comment (zeldin) [reply depends/check-ps3dev.sh]: Even assuming that was true (it could still be write protected), you don't actually know whether or not the owner of the folder is the effective user.  The folder could exist but be owned by someone else (root for example).

File: scripts/001-binutils-PPU.sh (hunk 4 above)
Comment (zeldin) [reply scripts/001-binutils-PPU.sh]: Well, I don't think making the build 8 seconds slower warrants adding the requirement to find and install a new tool.  (Not everyone runs Ubuntu or macOS.)

Comment (luizfernandonb) [reply scripts/001-binutils-PPU.sh]: You are looking at a simple file, you have to look at the entire toolchain, seeing this, it is clear that the extraction of `GCC 7.2.0` and `binutils 2.22` are the most time consuming of the entire toolchain, and certainly a program to extract these files using multiple threads will make a good difference in the complete installation time of the toolchain

My PC has a Pentium E6700 from 2010, so it doesn't make much difference, but for those who extract on an i7 or i5, you may notice the difference

**Comment by luizfernandonb** [comment]
After many adjustments and a long time waiting to finish the builds, I believe that the best I can extract from the toolchain build, that's it, I accept suggestions

File: scripts/001-binutils-PPU.sh (hunk 4 above)
Comment (zeldin) [reply scripts/001-binutils-PPU.sh]: Huh?  _Compiling_ gcc is what takes time.  Unpacking has got to be <1% of the entire toolchain build time.  If you have to dig up a nonstandard tool and compile and install it, that's going to take several minutes, which is unlikely to make it a win even if your computer is really slow and your internet connection is really fast.

On my computer (POWER9) the extraction of `gcc-7.2.0.tar.xz` takes 6 seconds.  CPU load is 131.3%, so one hardware thread is running `xz`, one is running `tar`, and the remaining 62 are idle.  Now the _compliation_ of gcc, that takes 4 minutes 42 seconds even though it can use all the threads.  So of the total time for extracting and compiling gcc, the extracting amounted for 2% of the time.  This ratio would stay the same even with a slower CPU (because both unpacking and compilation would be slower), but on a CPU with fewer threads the extraction would amount to even less because the compilation would be slower but not the extraction (which can only use 2 threads anyway).  So assuming the CPU has only 32 or less threads I'd say my guesstimate of <1% hits the mark.  😄 

Downloading gcc as `.gz` takes me 3 minutes 24 seconds.  Downloading gcc as `.xz` takes 1 minute 55 seconds.  So it saves more than 40% of the download time, or a minute and a half in absolute terms.  Yes, this is over an IPv6 tunnel, but if we assume slow computers we should assume slow connections too (and not necessarily in combination).

At the end of the day I guess your mileage will vary, but that's also why I wouldn't accept this as a mandatory one-size-fits all solution.  If you were to make it as an option for those who would both actually benefit and care enough to invest in getting this tool, then maybe...

Comment (luizfernandonb) [reply scripts/001-binutils-PPU.sh]: Ok, I will remove the `pigz` from the source code, I ended up agreeing with your explanation

Comment (luizfernandonb) [reply scripts/001-binutils-PPU.sh]: But we also have to look at people who do not have a PC as good as yours, here, unpacking the GCC takes a long time, and the compilation too, adding the unpacking/compilation for the PPU and SPU, here it takes at least 2 hours to complete (without taking into account binutils and GDB)

Comment (luizfernandonb) [reply scripts/001-binutils-PPU.sh]: Done, if you can review this PR, I will be grateful 🤝

Comment (zeldin) [reply scripts/001-binutils-PPU.sh]: Out of interrest, did you also measure how much of those 2 hours is spent on the actual uncompression?  With and without the pigz?  If you measure using `time` be careful that it only measures one process and that you have a pipeline in the pigz case but not in the regular one.  To get a fair measurement you can change the regular case to also be a pipeline and put a `time` on the `xz -dc` process to race it against `pigz -dc`.

File: README.md [hunk 7]
```diff
@@ -1,52 +1,60 @@
...
+videogame system.
 
-  This program will automatically build and install a compiler and other
-  tools used in the creation of homebrew software for the Sony PlayStation 3
-  videogame system.
+## How do I use it?
 
-  How do I use it?
- ==================
+1. If you are using Linux or macOS, add this to the login script (`~/.bashrc` on Linux)
```
Comment (zeldin) [inline README.md:10]: And if not?  This wording seems to suggest that nothing needs to be done in that case.  While the old wording might not have given you the correct details for every such case, it did at least convey that you had to set the variables.

I like the wording in your "2." better, which tells you what needs to be done and then notes that the concrete example may be relevant only to Linux.  So maybe we can say something like:

> Set the PS3DEV and PS1LIGHT environment variables, and the executable path (Linux `~/.bashrc` example)

?

File: README.md [hunk 8]
```diff
@@ -1,52 +1,60 @@
...
 
-## Linux
+  export PATH=$PATH:$PS3DEV/bin
+  export PATH=$PATH:$PS3DEV/ppu/bin
+  export PATH=$PATH:$PS3DEV/spu/bin
+```
 
-  Specifically on debian-based systems, the following command line should
-  be enough to install everything necessary:
+2. Give yourself permission to write files to the folder yourself (Linux example)
```
Comment (zeldin) [inline README.md:21]: Is it just me or does the second "yourself" here seem really superfluous?

File: README.md [hunk 9]
```diff
@@ -1,52 +1,60 @@
...
 
- 2) Add the following to your login script:
+#### macOS
+
+- On macOS systems, if you have [Homebrew](http://brew.sh) package manager, the following command line should be enough to install everything necessary:
+
 ```bash
-  export PS3DEV=/usr/local/ps3dev
-  export PSL1GHT=$PS3DEV
+brew install autoconf automake openssl libelf ncurses zlib gmp wget pkg-config pigz lbzip2
```
Comment (zeldin) [inline README.md:46]: Still has pigz here.  And I suppose lbzip2 is related to that as well?

File: README.md [hunk 10]
```diff
@@ -1,52 +1,60 @@
...
 ```bash
-  apt-get install autoconf automake bison flex gcc libelf-dev make \
-    texinfo libncurses5-dev patch python subversion wget zlib1g-dev \
-    libtool libtool-bin python-dev bzip2 libgmp3-dev pkg-config g++ libssl-dev clang
+sudo mkdir $PS3DEV
+sudo chown -R $USER:$USER $PS3DEV
 ```
 
-## macOS
+3. Update the shell for it with the new variables (on Linux the command is `source ~/.bashrc`)
```
Comment (zeldin) [inline README.md:28]: This sounds a little bit weird.  Maybe

> Make sure the new variables are set in the shell (...)

instead?

File: depends/check-psl1ght.sh [hunk 11]
```diff
@@ -5,7 +5,7 @@
 if test ! $PSL1GHT; then { echo "ERROR: Set \$PSL1GHT before continuing."; exit 1; } fi
 
 ## Check for the $PSL1GHT directory.
-( ls -ld $PSL1GHT || mkdir -p $PSL1GHT ) 1>/dev/null 2>&1 || { echo "ERROR: Create $PSL1GHT before continuing."; exit 1; }
+#( ls -ld $PSL1GHT || mkdir -p $PSL1GHT ) 1>/dev/null 2>&1 || { echo "ERROR: Create $PSL1GHT before continuing."; exit 1; }
```
Comment (zeldin) [inline depends/check-psl1ght.sh:8]: Don't remove the sanity check.  Removing the `|| mkdir -p $PSL1GHT` is ok since the README now says you have to create it yourself.
(And if we are changing this anyway I'd prefer `test -d` over `ls -ld` for the check.  :smile:)

File: depends/check-psl1ght.sh [hunk 12]
```diff
@@ -5,7 +5,7 @@
...
 
 ## Check for write permission.
-touch $PSL1GHT/test.tmp 1>/dev/null 2>&1 || { echo "ERROR: Grant write permissions for $PSL1GHT before continuing."; exit 1; }
+#touch $PSL1GHT/test.tmp 1>/dev/null 2>&1 || { echo "ERROR: Grant write permissions for $PSL1GHT before continuing."; exit 1; }
```
Comment (zeldin) [inline depends/check-psl1ght.sh:11]: As above, don't remove the sanity check.  It _would_ make sense, however, to `rm` the `test.tmp` file afterwards.

File: patches/binutils-2.22-PS3.patch [hunk 13]
```diff
@@ -12,4 +12,4 @@ diff -burN '--exclude=.git' binutils-2.22/ld/emultempl/spuelf.em binutils-2.22-P
 +#endif
    perror (argv[0]);
    _exit (127);
- }
+ }
```
Comment (zeldin) [inline patches/binutils-2.22-PS3.patch:15]: This looks like a mistake?

File: scripts/001-binutils-PPU.sh [hunk 14]
```diff
@@ -13,7 +15,7 @@ if [ ! -d ${BINUTILS} ]; then
   if [ ! -f config.sub ]; then wget --continue https://git.savannah.gnu.org/cgit/config.git/plain/config.sub; fi
 
   ## Unpack the source code.
-  tar xfvj ${BINUTILS}.tar.bz2
+  tar -xvjf ${BINUTILS}.tar.bz2
```
Comment (zeldin) [inline scripts/001-binutils-PPU.sh:18]: Don't add a dash here, the variant without is more portable.

File: scripts/001-binutils-PPU.sh [hunk 15]
```diff
@@ -34,18 +36,11 @@ fi
...
-    --disable-debug \
-    --disable-dependency-tracking \
-    --disable-werror \
-    --enable-64-bit-bfd \
-    --with-gcc \
-    --with-gnu-as \
-    --with-gnu-ld
+CFLAGS="-O3" CXXFLAGS="-O3" CFLAGS_FOR_TARGET="-O3" CXXFLAGS_FOR_TARGET="-O3" BOOT_CFLAGS="-O3" GOCFLAGS_FOR_TARGET="-O3" \
+../configure --prefix="$PS3DEV/ppu" --target="powerpc64-ps3-elf" --disable-nls --disable-shared --disable-debug --disable-dependency-tracking \
+             --disable-werror --enable-64-bit-bfd --with-gcc --with-gnu-as --with-gnu-ld --with-cpu="cell" --with-tune="cell" \
```
Comment (zeldin) [inline scripts/001-binutils-PPU.sh:41]: Hm, I feel that it was easier to read when the options were spread out on more lines, but I don't have a strong opinion here.

File: scripts/002-gcc-newlib-PPU.sh [hunk 16]
```diff
@@ -11,8 +13,8 @@ if [ ! -d ${GCC} ]; then
   if [ ! -f ${NEWLIB}.tar.gz ]; then wget --continue https://sourceware.org/pub/newlib/${NEWLIB}.tar.gz; fi
 
   ## Unpack the source code.
-  rm -Rf ${GCC} && tar xfvJ ${GCC}.tar.xz
-  rm -Rf ${NEWLIB} && tar xfvz ${NEWLIB}.tar.gz
+  rm -Rf ${GCC} && tar -xvJf ${GCC}.tar.xz
+  rm -Rf ${NEWLIB} && tar -xvf ${NEWLIB}.tar.gz
```
Comment (zeldin) [inline scripts/002-gcc-newlib-PPU.sh:17]: Again, no dash.
You can remove the `v` if you want; the printing of the filenames can actually take some time on a slow system and it doesn't really serve any purpose.

File: scripts/003-gdb-PPU.sh [hunk 17]
```diff
@@ -1,19 +1,21 @@
 #!/bin/sh -e
+#
 # gdb-PPU.sh by Naomi Peori (naomi@peori.ca)
+# Modified by luizfernandonb (luizfernando.nb@outlook.com)
 
-GDB="gdb-7.5.1"
+GDB="gdb-10.1"
```
Comment (zeldin) [inline scripts/003-gdb-PPU.sh:6]: Did you test gdb 10.1 that it works?  Or only that it compiles?

File: scripts/003-gdb-PPU.sh [hunk 18]
```diff
@@ -1,19 +1,21 @@
...
-  if [ ! -f ${GDB}.tar.bz2 ]; then wget --continue https://ftp.gnu.org/gnu/gdb/${GDB}.tar.bz2; fi
+  if [ ! -f ${GDB}.tar.xz ]; then wget --continue https://ftp.gnu.org/gnu/gdb/${GDB}.tar.xz; fi
 
   ## Download an up-to-date config.guess and config.sub
   if [ ! -f config.guess ]; then wget --continue https://git.savannah.gnu.org/cgit/config.git/plain/config.guess; fi
   if [ ! -f config.sub ]; then wget --continue https://git.savannah.gnu.org/cgit/config.git/plain/config.sub; fi
 
   ## Unpack the source code.
-  tar xfvj ${GDB}.tar.bz2
+  tar -xvJf ${GDB}.tar.xz
```
Comment (zeldin) [inline scripts/003-gdb-PPU.sh:18]: No dash.

File: scripts/005-binutils-SPU.sh [hunk 19]
```diff
@@ -13,7 +15,7 @@ if [ ! -d ${BINUTILS} ]; then
   if [ ! -f config.sub ]; then wget --continue https://git.savannah.gnu.org/cgit/config.git/plain/config.sub; fi
 
   ## Unpack the source code.
-  tar xfvj ${BINUTILS}.tar.bz2
+  tar -xvjf ${BINUTILS}.tar.bz2
```
Comment (zeldin) [inline scripts/005-binutils-SPU.sh:18]: No dash.

File: scripts/006-gcc-newlib-SPU.sh [hunk 20]
```diff
@@ -11,8 +13,8 @@ if [ ! -d ${GCC} ]; then
   if [ ! -f ${NEWLIB}.tar.gz ]; then wget --continue https://sourceware.org/pub/newlib/${NEWLIB}.tar.gz; fi
 
   ## Unpack the source code.
-  rm -Rf ${GCC} && tar xfvJ ${GCC}.tar.xz
-  rm -Rf ${NEWLIB} && tar xfvz ${NEWLIB}.tar.gz
+  rm -Rf ${GCC} && tar -xvJf ${GCC}.tar.xz
+  rm -Rf ${NEWLIB} && tar -xvf ${NEWLIB}.tar.gz
```
Comment (zeldin) [inline scripts/006-gcc-newlib-SPU.sh:17]: No dash.

File: scripts/007-gdb-SPU.sh [hunk 21]
```diff
@@ -1,19 +1,21 @@
...
-  if [ ! -f ${GDB}.tar.bz2 ]; then wget --continue https://ftp.gnu.org/gnu/gdb/${GDB}.tar.bz2; fi
+  if [ ! -f ${GDB}.tar.xz ]; then wget --continue https://ftp.gnu.org/gnu/gdb/${GDB}.tar.xz; fi
 
   ## Download an up-to-date config.guess and config.sub
   if [ ! -f config.guess ]; then wget --continue https://git.savannah.gnu.org/cgit/config.git/plain/config.guess; fi
   if [ ! -f config.sub ]; then wget --continue https://git.savannah.gnu.org/cgit/config.git/plain/config.sub; fi
 
   ## Unpack the source code.
-  tar xfvj ${GDB}.tar.bz2
+  tar -xvJf ${GDB}.tar.xz
```
Comment (zeldin) [inline scripts/007-gdb-SPU.sh:18]: No dash.

File: scripts/008-psl1ght.sh [hunk 22]
```diff
@@ -1,11 +1,13 @@
...
+#
 # psl1ght.sh by Naomi Peori (naomi@peori.ca)
+# Modified by luizfernandonb (luizfernando.nb@outlook.com)
 
 ## Download the source code.
 wget --no-check-certificate https://github.com/ps3dev/PSL1GHT/tarball/master -O psl1ght.tar.gz
 
 ## Unpack the source code.
-rm -Rf psl1ght && mkdir psl1ght && tar --strip-components=1 --directory=psl1ght -xvzf psl1ght.tar.gz
+rm -Rf psl1ght && mkdir psl1ght && pigz -dc psl1ght.tar.gz | tar --strip-components=1 --directory=psl1ght -xvf -
```
Comment (zeldin) [inline scripts/008-psl1ght.sh:10]: No pigz.

File: scripts/009-ps3libraries.sh [hunk 23]
```diff
@@ -1,11 +1,14 @@
...
+#
 # ps3libraries.sh by Naomi Peori (naomi@peori.ca)
+# Modified by luizfernandonb (luizfernando.nb@outlook.com)
 
 ## Download the source code.
 wget --no-check-certificate https://github.com/ps3dev/ps3libraries/tarball/master -O ps3libraries.tar.gz
 
 ## Unpack the source code.
-rm -Rf ps3libraries && mkdir ps3libraries && tar --strip-components=1 --directory=ps3libraries -xvzf ps3libraries.tar.gz && cd ps3libraries
+rm -Rf ps3libraries && mkdir ps3libraries && pigz -dc ps3libraries.tar.gz | tar --strip-components=1 --directory=ps3libraries -xvf \
```
Comment (zeldin) [inline scripts/009-ps3libraries.sh:10]: No pigz.

**Review by zeldin — CHANGES_REQUESTED** [review]
We're getting there...  😄

File: patches/binutils-2.22-PS3.patch (hunk 13 above)
Comment (luizfernandonb) [reply patches/binutils-2.22-PS3.patch:15]: No, it's the code itself, [see here](https://github.com/ps3dev/ps3toolchain/blob/master/patches/binutils-2.22-PS3.patch)

**Comment by miigotu** [comment]
I just realized this is seriously going to conflict with my PR lol

File: patches/binutils-2.22-PS3.patch (hunk 13 above)
Comment (zeldin) [reply patches/binutils-2.22-PS3.patch:15]: ?  You removed the final newline from the patch file.  The corresponding "}" line in `ld/emultemp/spuelf.em` in binutils-2.22 certainly does end with a newline (it's not even the last line of the file), so it's incorrect for the context line in the patch to be lacking one.  I don't understand what you mean with "it" (whatever that refers to) being "the code itself"...

File: scripts/003-gdb-PPU.sh (hunk 17 above)
Comment (luizfernandonb) [reply scripts/003-gdb-PPU.sh:6]: I couldn't test it, I tried to follow the steps in [this tutorial](https://sourceware.org/gdb/current/onlinedocs/gdb/Connecting.html#:~:text=In%20target%20remote%20mode%2C%20if,GDB%20connected%20to%20the%20target.) and I couldn't because I don't know the port of my PS3, if you can help me how to test the GDB, I'll be grateful, but I'm 90% sure it will work because of this [here](https://stackoverflow.com/questions/29801755/c-do-i-need-gcc-and-gdb-with-the-same-version-to-debug)

Comment (luizfernandonb) [reply scripts/003-gdb-PPU.sh:6]: Do you know what is the PS3 port for GDB to connect to?

**Comment by luizfernandonb** [comment]
@shagkur @zeldin Here are the requested changes (I believe), and sorry for the delay, my college class has started, I ask you to review this latest commit

File: README.md [hunk 24]
```diff
@@ -1,52 +1,64 @@
...
-  export PSL1GHT=$PS3DEV
+autoconf automake openssl libelf ncurses zlib gmp wget pkg-config
+```
 
-  export PATH=$PATH:$PS3DEV/bin
-  export PATH=$PATH:$PS3DEV/ppu/bin
-  export PATH=$PATH:$PS3DEV/spu/bin
+### Install toolchain
+
+3. Run the toolchain script:
```
Comment (zeldin) [inline README.md:55]: This should now be 5. due to the added steps above.  Or maybe 6. if you want to make the install dependencies step number 5 (it used to be 1., but now it appears to be unnumbered).

File: patches/binutils-2.22-PS3.patch (hunk 13 above)
Comment (zeldin) [reply patches/binutils-2.22-PS3.patch:15]: This change is still here.  I don't think it should be.  If you don't know how to undo it here is how:
```
git checkout origin/master -- patches/binutils-2.22-PS3.patch
git commit -m 'Restore binutils patch' -- patches/binutils-2.22-PS3.patch
```
Then the change will go away when we squash the commits.

**Review by zeldin — COMMENTED** [review]
Hi.  Thanks for hanging in there.  :smile:
Now there are only two small nits left, one with the numbering in the README, and one with the bogus change to the binutils patch.
Do you think you could rebase this on the current master, sorting out the conflicts?
Thanks

**Comment by luizfernandonb** [comment]
@zeldin I made the changes you asked for and executed the command that you sent me, please see if you had the result waiting, when I executed the command here, it didn't show any errors, but I also didn't see a success message

And about numbering, I decided to do it as if it were subprocesses, for example, step `4` if you have any peculiarities, the next number is `4.1` and etc.

**Review by zeldin — APPROVED** [review]
Ok.  It seems a little bit strange though to have lists of packages to install as "subprocess" to step 4, since step 4 is just "Make sure the new variables are set in the shell", and has nothing to do with installing packages...

But the main thing needed now is to rebase this on master and fix the conflicts.  You probably want to squash everything to a single commit first.

The issue with the patch file still seems to be there, but we can sort that out after you rebase.
